src/plugins/azgenerator/TemplateAzureCliParams.ts

```diff
@@ -145,8 +145,8 @@ function getCommandBody(model: CodeModelAz, needUpdate: boolean = false) {
                         argument += ", help='" + EscapeString(model.MethodParameter_Description) + "'";
                     }
         
-                    if (model.MethodParameter_IsList) {
-                        if (model.MethodParameter_Type == SchemaType.Object || model.MethodParameter_Type == SchemaType.Array) {
+                    if (model.Option_IsList) {
+                        if (model.Option_IsListOfComplex) {
```

Comment:
This is not correct to use the Option_* . Because the outside loop is looping for methods. You should have MethodParameter_IsList and MethodParameter_IsListOfComplex.

Reply:
And allParam is used to make sure the same parameters of methods in the same command will only appear once.

Reply:
ok, i will change it, i have to look into that more thoroughly, as the implementation of these looks right now very similar